src/vmm/src/lib.rs

```diff
@@ -684,15 +688,15 @@ impl Vmm {
     pub fn flush_metrics(&mut self) -> UserResult {
         self.write_metrics().map_err(|e| {
             let (kind, error_contents) = match e {
-                LoggerError::NeverInitialized(s) => (ErrorKind::User, s),
+                MetricsError::NeverInitialized(s) => (ErrorKind::User, s),
                 _ => (ErrorKind::Internal, e.to_string()),
             };
-            VmmActionError::Logger(kind, LoggerConfigError::FlushMetrics(error_contents))
+            VmmActionError::Metrics(kind, MetricsConfigError::FlushMetrics(error_contents))
         })
     }
 
-    fn write_metrics(&mut self) -> result::Result<(), LoggerError> {
-        LOGGER.log_metrics().map(|_| ())
+    fn write_metrics(&mut self) -> result::Result<(), MetricsError> {
```

Comment:
There is a mismatch between the method name and what it seems it does. I think it is the result of having `write` and `flush` operations combined almost everywhere when it comes to logger and metrics. If we really need to combine them, I think it is better if we keep the `flush` part optional through a parameter, when it comes to `writing`.

Reply:
I don't know if it makes sense to keep the `flush` part optional if we want, for our particular use case, to always call `flush` after `write`. But I agree the naming here was not so clear, I renamed this function `flush_metrics` and the one that is called when user sends `FlushMetrics` action -> `force_flush_metrics`.